src/test/run-make-fulldeps/target-cpu-native/Makefile

```diff
@@ -1,5 +1,7 @@
 -include ../tools.mk
 
 all:
-	$(RUSTC) foo.rs -C target-cpu=native
+	$(RUSTC) foo.rs -C target-cpu=native 2>&1 | tee $(TMPDIR)/out.log
 	$(call RUN,foo)
+	# Make sure no warnings about "unknown CPU `native`" were emitted
+	if grep 'is not a recognized processor' $(TMPDIR)/out.log; then exit 1; fi
```

Comment:
I would prefer to check that no warnings at all are emitted (i.e. that `out.log` is empty), to avoid this recurring with a different error message.

I don't think the compiler should ever emit garbage when compiling an empty file.

Reply:
Are there platforms where the compiler is always chatty? If then, I would check that the compiler is not chatty on all the other platforms.

Reply:
I basically hate writing tests in bash, do you know how to test a file is empty?

Reply:
Maybe use ````wc```` and count the number of characters?
````wc --chars file | cut -d' ' -f1```` should be 0.

Reply:
How do you test against 0 though? (I'm really bad at bash...)

Reply:
````bash
#!/bin/bash
touch file

if [ $(wc --chars file | cut -d' ' -f1) == 0 ]; then
	echo "it's zero!"
else
	echo "it's not zero!"
fi
````
should work :)

Reply:
Wouldn't `-s FILE` work? "True if FILE exists and has a size greater than zero."